src/utils/feeEstimate.js

```diff
@@ -1,37 +1,69 @@
-import {ethers} from "ethers";
+import { ethers } from "ethers";
 import * as veFeeEstimateABI from "./abis/veFeeEstimateABI";
-import {networkSigner} from "../stores/web3";
-import {getAddressByChainIdKey} from "../utils/address/address";
+import {
+  prepareWriteContract,
+  readContract,
+  waitForTransaction,
+  writeContract,
+} from "@wagmi/core";
+import { getAddressByChainIdKey } from "../utils/address/address";
+import { getGasFeeEstimate } from "./web3";
 
-export const getRewardsFeeEstimate = async(userAddress, provider) => {
-    try {
-      const contract = new ethers.Contract(
-        getAddressByChainIdKey(process.env.VE_SUPPORTED_CHAINID, "veFeeEstimate"),
-        veFeeEstimateABI.default, 
-        provider
-      );
-      const estimateClaim = await contract.estimateClaim(userAddress)
-      const estimateClaimFormatted = ethers.utils.formatEther(BigInt(estimateClaim).toString(10))
-      return estimateClaimFormatted
+export const getRewardsFeeEstimate = async (userAddress) => {
+  try {
+    const estimateClaim = await readContract({
+      address: getAddressByChainIdKey(
+        import.meta.env.VITE_VE_SUPPORTED_CHAINID,
+        "veFeeEstimate"
+      ),
+      args: [userAddress],
+      abi: veFeeEstimateABI.default,
+      functionName: "estimateClaim",
+    });
+    const estimateClaimFormatted = ethers.utils.formatEther(
+      BigInt(estimateClaim).toString(10)
+    );
+    return estimateClaimFormatted;
   } catch (error) {
-    console.log(error)
+    console.log(error);
     throw error;
   }
-}
+};
 
-export async function claimVERewards(userAddress, signer) {
+export async function claimVERewards(userAddress) {
   try {
```

Comment:
Did not test latest implementation of claim functions inside in-app/test environment, only accessible functions in local goerli. Have reviewed it and implementation looks valid. Will continue with testing app throughout this week.